docs/conf/user-interface/index.md

```diff
@@ -0,0 +1,621 @@
+sidebar_label: Introduction
+description: Learn how to customize the look and feel of your Backstage app, including theming and branding options.
+---
+
+Backstage offers built-in support for both light and dark themes, making it easy to get started with a professional look and feel. But many teams want to go further—tailoring the interface to reflect their organization’s unique brand, identity, and experience.
+
+This section explores the different ways you can customize the appearance of your Backstage instance. You'll learn how the theming system is structured today, how to work with the two coexisting UI systems, and how to define themes that align with your visual language.
+
+## Theming architecture overview
+
+Backstage currently supports two parallel UI systems. The original theming and component model is built on Material UI (MUI), a popular React-based framework. More recently, Backstage introduced Backstage UI (BUI), a custom-designed, CSS-first system developed to meet the platform’s evolving needs. Both systems are supported today, with many parts of the ecosystem still using MUI while new components adopt BUI.
+
+<div style="display: flex; gap: 1rem; flex-wrap: wrap; margin-bottom: 1rem;">
+  <div style="flex: 1; min-width: 250px; border: 1px solid #ccc; border-radius: 8px; padding: 1rem;">
+    <h3 style="margin-top: 0;">MUI (Legacy)</h3>
+    <ul>
+      <li><strong>Theming:</strong> JS-based with <code>UnifiedThemeProvider</code></li>
+      <li><strong>Coverage:</strong> Most existing plugins</li>
+      <li><strong>Documentation:</strong> <a href="https://mui.com/material-ui/" target="_blank">mui.com</a></li>
+    </ul>
+  </div>
+  <div style="flex: 1; min-width: 250px; border: 1px solid #ccc; border-radius: 8px; padding: 1rem;">
+    <h3 style="margin-top: 0;">Backstage UI (New)</h3>
+    <ul>
+      <li><strong>Theming:</strong> CSS variables and tokens</li>
+      <li><strong>Coverage:</strong> Growing, focused on new work</li>
+      <li><strong>Documentation:</strong> <a href="https://ui.backstage.io" target="_blank">ui.backstage.io</a></li>
+    </ul>
+  </div>
+</div>
+
+:::info
+We recognize that maintaining two separate theming systems is not ideal. Because of the fundamental architectural differences between MUI and Backstage UI, it can be challenging to automate theme updates or know exactly which theme to modify for a given component. Our recommendation is to inspect the component’s code and check its class names: if you see a class name starting with `bui`, you should use the Backstage UI theming approach to style it.
+:::
+
+## Creating custom themes
+
+During the transition to Backstage UI, you will need to maintain themes in two places: some components and plugins still rely on MUI, while others use Backstage UI. At this time, there is no automated solution to generate MUI themes from your Backstage UI theme, so you will need to manage both separately until the migration is complete.
```

Comment:
Solution on the way in https://github.com/backstage/backstage/pull/31140. Perhaps we can mention that this is in progress and refer to the PR for now? Thinking we want to ship these docs way sooner than the plugin

Reply:
I just updated the text to mention our progress on that front with a link to your PR.